express-mutations/server.ts

```diff
@@ -0,0 +1,117 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+// Endpoint for testing
+app.get('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+    if (!Number.isInteger(+actorId)) {
+      throw new ClientError(400, `Non-integer actorId: ${actorId}`);
+    }
+    const sql = `
+      select * from "actors"
+      where "actorId" = $1;
+    `;
+    const params = [actorId];
+    const result = await db.query(sql, params);
+    const actor = result.rows[0];
+    if (!actor) throw new ClientError(404, `actor ${actorId} not found`);
+    res.json(actor);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    const { firstName, lastName } = req.body;
+    if (!firstName || !lastName) {
+      throw new ClientError(400, 'First and last name required');
+    }
+    const sql = `
+    insert into "actors" ("firstName", "lastName")
+    values ($1, $2)
+    returning *;
+    `;
+    const params = [firstName, lastName];
+    const result = await db.query(sql, params);
+    const actor = result.rows[0];
+    res.status(201).json(actor);
+  } catch (err) {
+    next(err);
+  }
+});
+
+// http POST localhost:8080/api/actors firstName="Brendan" lastName="Eich"
+
+app.put('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+    const { firstName, lastName } = req.body;
+    if (!actorId || !firstName || !lastName) {
+      throw new ClientError(404, 'Actor Id, First and Last name required');
```

Comment:
Client errors in the request should be 400. A 404 is just for when they ask for an ID that doesn't exist.